stl/inc/functional

```diff
@@ -1207,6 +1207,604 @@ _NODISCARD bool operator!=(nullptr_t, const function<_Fty>& _Other) noexcept {
+    }
+};
+
+template <class _Rx, class... _Types>
+[[noreturn]] _Rx _Function_no_callable(const _Move_only_function_data&, _Types&&...) noexcept {
+    _CSTD abort(); // We are not std::function to throw bad_function_call
+}
+
+template <class _Vt, class _VtInvQuals, class _Rx, class... _Types>
+_NODISCARD inline _Rx _Function_invoke_small(const _Move_only_function_data& _Self, _Types&&... _Args) {
+    return _Invoker_ret<_Rx>::template _Call<_VtInvQuals>(*_Self._Small_fn_ptr<_Vt>(), _STD forward<_Types>(_Args)...);
+}
+
+template <class _Vt, class _VtInvQuals, class _Rx, class... _Types>
+_NODISCARD inline _Rx _Function_invoke_large(const _Move_only_function_data& _Self, _Types&&... _Args) {
+    return _Invoker_ret<_Rx>::template _Call<_VtInvQuals>(*_Self._Large_fn_ptr<_Vt>(), _STD forward<_Types>(_Args)...);
+}
+
+template <class _Vt>
+inline void _Function_move_small(_Move_only_function_data& _Self, _Move_only_function_data& _Src) noexcept {
+    ::new (_Self._Buf_ptr<_Vt>()) _Vt(_STD move(*_Src._Small_fn_ptr<_Vt>()));
+    _Src._Small_fn_ptr<_Vt>()->~_Vt();
+    _Self._Impl = _Src._Impl;
+}
+
+inline void _Function_move_large(_Move_only_function_data& _Self, _Move_only_function_data& _Src) noexcept {
+    _CSTD memcpy(&_Self._Data, &_Src._Data, 2 * sizeof(void*)); // Copy Impl* and functor data
+}
+
+template <class _Vt>
+inline void _Function_destroy_small(_Move_only_function_data& _Self) noexcept {
+    _Self._Small_fn_ptr<_Vt>()->~_Vt();
+}
+
+inline void _Function_deallocate_large_default_aligned(_Move_only_function_data& _Self) noexcept {
+    ::operator delete(_Self._Large_fn_ptr<void>());
+}
+
+template <size_t _Align>
+inline void _Function_deallocate_large_overaligned(_Move_only_function_data& _Self) noexcept {
+    ::operator delete (_Self._Large_fn_ptr<void>(), align_val_t{_Align});
+}
+
+template <class _Vt>
+inline void _Function_destroy_large(_Move_only_function_data& _Self) noexcept {
+    auto* _Pfn = _Self._Large_fn_ptr<_Vt>();
+    _Pfn->~_Vt();
+    if constexpr (alignof(_Vt) <= __STDCPP_DEFAULT_NEW_ALIGNMENT__) {
+        ::operator delete(static_cast<void*>(_Pfn));
+    } else {
+        ::operator delete (static_cast<void*>(_Pfn), align_val_t{alignof(_Vt)});
+    }
+}
+
+template <size_t _Size>
+inline void _Function_move_memcpy(_Move_only_function_data& _Self, _Move_only_function_data& _Src) noexcept {
+    _CSTD memcpy(&_Self._Data, &_Src._Data, _Size); // Copy Impl* and functor data
+}
+
+template <class _Vt>
+inline constexpr size_t _Function_small_copy_size = // We copy Impl* and the functor data at once
+    _Move_only_function_data::_Buf_offset<_Vt> + // Impl* plus possible alignment
+    (size_t{sizeof(_Vt) + sizeof(void*) - 1} & ~size_t{sizeof(void*) - 1}); // size in whole pointers
+
+template <class _Vt, class... _CTypes>
+_NODISCARD inline void* _Function_new_large(_CTypes&&... _Args) {
+    struct _NODISCARD _Guard_type {
+        void* _Ptr;
+
+        ~_Guard_type() {
+            if (_Ptr) {

```

Comment:
Probably I should comment this. `nullptr` here is very likely, it means that functor constructor did no throw. I want to avoid dll call of `::operator delete`.

Reply:
Commented